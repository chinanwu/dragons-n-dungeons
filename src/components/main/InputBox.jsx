import PropTypes from 'prop-types';
import React from 'react';

import Input from '../general/Input.jsx';

import './InputBox.less';

export const InputBox = ({
  id,
  className,
  label,
  value,
  type,
  onChange,
  onKeyDown,
}) => (
  <>
    <div id={`${id}Label`} className="InputBox__label">
      {label}
    </div>
    <Input
      id={id}
      className={className}
      type={type}
      value={value}
      ariaLabelledBy={`${id}Label`}
      onChange={onChange}
      onKeyDown={onKeyDown}
    />
  </>
);

InputBox.propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
};

export default InputBox;
