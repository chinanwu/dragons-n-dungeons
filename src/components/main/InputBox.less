.InputBox__label {
  text-align: center;
  margin-top: 6px;
}
