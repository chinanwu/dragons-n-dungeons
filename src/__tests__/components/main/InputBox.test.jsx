import { shallow } from 'enzyme';
import React from 'react';

import { InputBox } from '../../../components/main/InputBox.jsx';

jest.unmock('../../../components/main/InputBox.jsx');

describe('InputBox component', () => {
  describe('rendering', () => {
    it('renders InputBox', () => {
      const wrapper = shallow(<InputBox id="test" />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('calls onChange when input text', () => {
      const onChange = jest.fn();
      const wrapper = shallow(<InputBox id="test" onChange={onChange} />);
      expect(wrapper.getElement()).toMatchSnapshot();
      wrapper.find('#test').simulate('change');
      expect(onChange).toHaveBeenCalled();
    });
  });
});
