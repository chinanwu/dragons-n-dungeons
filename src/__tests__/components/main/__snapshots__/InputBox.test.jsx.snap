// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox component behaviour calls onChange when input text 1`] = `
<React.Fragment>
  <div
    className="InputBox__label"
    id="testLabel"
  />
  <Input
    ariaLabelledBy="testLabel"
    id="test"
    onChange={[MockFunction]}
  />
</React.Fragment>
`;

exports[`InputBox component rendering renders InputBox 1`] = `
<React.Fragment>
  <div
    className="InputBox__label"
    id="testLabel"
  />
  <Input
    ariaLabelledBy="testLabel"
    id="test"
  />
</React.Fragment>
`;
